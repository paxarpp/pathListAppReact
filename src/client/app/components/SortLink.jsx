import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Button from '../components/Button';

export default class SortLink extends Component {


    render() {
        const { name, nameCell, reverse } = this.props;
        return (
            name === nameCell ? (
                <Button handler={null}
                    styleButton="sortView">
                    {reverse ? String.fromCharCode(9660) : String.fromCharCode(9650)}
                </Button>
            ) : null
        )
    }
}