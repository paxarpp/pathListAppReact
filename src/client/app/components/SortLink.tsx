import React from 'react';

interface IProp {
  name: string;
  nameCell: string;
  reverse: boolean;
}

import Icon from './Icon';

const SortLink = ({ name, nameCell, reverse }:IProp) =>
  name === nameCell ? (
    reverse ? (
      <Icon onClick={null} name="Arrow_Drop_Down" />
    ) : (
      <Icon onClick={null} name="Arrow_Drop_Up" />
    )
  ) : (
    <Icon onClick={null} name="UnfoldMore" />
  );

export default SortLink;
