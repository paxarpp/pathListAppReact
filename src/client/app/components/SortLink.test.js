import React from 'react';
import renderer from 'react-test-renderer';
import SortLink from './SortLink';

test('SortLink view, reverse true', () => {
  const nameCell = 'test';
  const reverse = true;
  const name = 'test';
  const component = renderer.create(<SortLink nameCell={nameCell} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('SortLink view, reverse false', () => {
  const nameCell = 'test';
  const reverse = false;
  const name = 'test';
  const component = renderer.create(<SortLink nameCell={nameCell} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('SortLink noview', () => {
  const nameCell = 'test';
  const reverse = false;
  const name = 'diff';
  const component = renderer.create(<SortLink nameCell={nameCell} reverse={reverse} name={name} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
