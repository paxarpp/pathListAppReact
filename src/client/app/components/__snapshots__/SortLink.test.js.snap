// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortLink noview 1`] = `
<div
  className="iconSort"
>
  <div>
    ▲
  </div>
  <div>
    ▼
  </div>
</div>
`;

exports[`SortLink view, reverse false 1`] = `
<a
  className="sortView"
  onClick={null}
>
  ▲
</a>
`;

exports[`SortLink view, reverse true 1`] = `
<a
  className="sortView"
  onClick={null}
>
  ▼
</a>
`;
